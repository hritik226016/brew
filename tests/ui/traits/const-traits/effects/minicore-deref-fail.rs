//@ aux-build:minicore.rs
//@ compile-flags: --crate-type=lib -Znext-solver

#![feature(no_core, const_trait_impl, effects)]
//~^ WARN the feature `effects` is incomplete
#![no_std]
#![no_core]

extern crate minicore;
use minicore::*;

struct Ty;
impl Deref for Ty {
    type Target = ();
    fn deref(&self) -> &Self::Target { &() }
}

const fn foo() {
    *Ty;
    //~^ ERROR the trait bound `Ty: ~const minicore::Deref` is not satisfied
}
