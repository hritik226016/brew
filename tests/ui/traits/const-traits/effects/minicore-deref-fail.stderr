warning: the feature `effects` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/minicore-deref-fail.rs:4:39
   |
LL | #![feature(no_core, const_trait_impl, effects)]
   |                                       ^^^^^^^
   |
   = note: see issue #102090 <https://github.com/rust-lang/rust/issues/102090> for more information
   = note: `#[warn(incomplete_features)]` on by default

error[E0277]: the trait bound `Ty: ~const minicore::Deref` is not satisfied
  --> $DIR/minicore-deref-fail.rs:19:5
   |
LL |     *Ty;
   |     ^^^

error: aborting due to 1 previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0277`.
